core/src/main/java/org/apache/calcite/sql/dialect/PostgresqlSqlDialect.java

```diff
@@ -75,6 +75,13 @@ public class PostgresqlSqlDialect extends SqlDialect {
             return super.getMaxPrecision(typeName);
           }
         }
+
+        @Override public int getDefaultPrecision(SqlTypeName typeName) {
```

Comment:
was the previous behavior incorrect?
CHAR(1) is a legal type.

Reply:
Related to the issue: https://issues.apache.org/jira/browse/CALCITE-6423
select cast('a' as char);
select cast('a' as char(1));
both of them return a;
and had a discuss in JIRA: https://issues.apache.org/jira/browse/CALCITE-6919 @mihaibudiu